boofuzz/blocks/checksum.py

```diff
@@ -121,9 +122,7 @@ def _should_render_fuzz_value(self):
         return self._fuzzable and (self._mutant_index != 0) and not self._fuzz_complete
 
     def _get_dummy_value(self):
-        if self._length:
-            return self._length * "\x00"
-        return self.checksum_lengths[self._algorithm] * "\x00"
+        return self._length * "\x00"
```

Comment:
We can simplify this as
https://github.com/jtpereyda/boofuzz/blob/c81f0d9232929fbbe534bb3c850a02e6b6f6ad2f/boofuzz/blocks/checksum.py#L77-L78
will assure that `self._length` will be updated with the correct value before `_get_dummy_value` is called.

If we use a custom function for the checksum algorithm and don't specify it's length, the dummy value will be empty. This is wrong but doesn't have an effect on fuzzing as far as I know. It is now also clearly documented, that `length` needs to be set when using a custom algorithm.